omod/src/main/java/org/openmrs/module/fhir2/providers/ConditionFhirResourceProvider.java

```diff
@@ -70,4 +80,21 @@ public Condition getConditionByUuid(@IdParam @NotNull IdType id) {
 	public MethodOutcome createCondition(@ResourceParam Condition newCondition) {
 		return FhirServerUtils.buildCreate(conditionService.saveCondition(newCondition));
 	}
+	
+	@Search
+	//@SuppressWarnings("unused")
+	public Bundle searchConditions(
+	        @OptionalParam(name = Condition.SP_PATIENT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_NAME,
+	                Patient.SP_GIVEN, Patient.SP_FAMILY }) ReferenceAndListParam patientParam,
+	        @OptionalParam(name = Condition.SP_SUBJECT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_NAME,
+	                Patient.SP_GIVEN, Patient.SP_FAMILY }) ReferenceAndListParam subjectParam,
+	        @OptionalParam(name = Condition.SP_CODE) TokenOrListParam code,
+	        @OptionalParam(name = Condition.SP_CLINICAL_STATUS) TokenOrListParam clinicalStatus,
+	        @OptionalParam(name = Condition.SP_ONSET_DATE) DateParam onsetDate,
```

Comment:
I would use a `DateRangeParam` here. `DateParam` only expresses a single date. `DateRangeParam` can express a single date or (more obviously) a point that is between two dates / instants.